nix/default.nix

```diff
@@ -2,20 +2,17 @@
 let
   nix-pre-commit-hooks = (import (builtins.fetchTarball "https://github.com/cachix/pre-commit-hooks.nix/tarball/master/" + "/nix/") { sources = sources; }).packages;
   overlay = _self: pkgs:
-    let sharedOverrides = {
-        overrides = _self: super: {
-            mkDerivation = args: super.mkDerivation (args //
-                {
-                    # skip running tests for Hackage packages
-                    doCheck = args.pname == "ghcide" || args.pname == "haskell-language-server";
-                    # relax upper bounds
-                    jailbreak = args.pname != "jailbreak-cabal";
-                });
-            };
-            };
-        gitignoreSource = (import sources.gitignore { inherit (pkgs) lib; }).gitignoreSource;
-        extended = haskellPackages:
-          haskellPackages.extend (pkgs.haskell.lib.packageSourceOverrides {
+    let
+        sharedOverrides = {
+          overrides = _self: super: {
+            mkDerivation = args:
+              let
+                broken = args.broken or false;
+                check = args.doCheck or true;
+              in super.mkDerivation (args // { jailbreak = broken; doCheck = if broken then false else check; });
```

Comment:
it would be good to have some comments describing the `mkDerivation` changes

Reply:
Two notes here:

1. If we bump nixpkgs very frequently this could sometimes break if there is a nixpkgs commit where a broken package is not marked as such.

2. I will register myself as maintainer in nixpkgs for all hls dependencies I know about so that it’s going to be very unlikely that we experience any breakage here.